build.sh

```diff
@@ -0,0 +1,130 @@
+#!/bin/bash
+
+# Exit if any error occurs
+set -e
+
+#
+# Display a help message.
+function displayHelp() {
+    echo "This script helps you build the syndesis monorepo."
+    echo "The available options are:"
+    echo " --skip-tests          Skips the test execution."
+    echo " --skip-image-builds   Skips image builds."
+    echo " --clean               Cleans up the projects."
+    echo " --namespace N         Specifies the namespace to use."
+    echo " --help                Displays this help message."
+}
+
+#
+# Checks if a flag is present in the arguments.
+function hasflag() {
+    filter=$1
+    for var in "${@:2}"; do
+        if [ "$var" = "$filter" ]; then
+            echo 'true'
+            break;
+        fi
+    done
+}
+
+#
+# Read the value of an option.
+function readopt() {
+        filter=$1
+        next=false
+        for var in "${@:2}"; do
+                if $next; then
+                        echo $var
+                        break;
+                fi
+                if [ "$var" = "$filter" ]; then
+                        next=true
+                fi
+        done
+}
+
+#
+# Options and flags
+SKIP_TESTS=$(hasflag --skip-tests "$@" 2> /dev/null)
+SKIP_IMAGE_BUILDS=$(hasflag --skip-image-builds "$@" 2> /dev/null)
+CLEAN=$(hasflag --clean "$@" 2> /dev/null)
+
+NAMESPACE=$(readopt --namespace "$@" 2> /dev/null)
+
+HELP=$(hasflag --help "$@" 2> /dev/null)
+
+
+if [ -n "$HELP" ]; then
+   displayHelp
+   exit 0
+fi
+
+#
+# Internal variable default values
+OC_OPTS=""
+MAVEN_OPTS=""
+MAVEN_CLEAN_GOAL="clean"
+MAVEN_IMAGE_BUILD_GOAL="fabric8:build"
+
+#
+# Apply options
+if [ -n "$SKIP_TESTS" ]; then
+    echo "Skipping tests ..."
+    MAVEN_OPTS="$MAVEN_OPTS -DskipTests"
+fi
+
+if [ -n "$SKIP_IMAGE_BUILDS" ]; then
+    echo "Skipping image builds ..."
+    MAVEN_IMAGE_BUILD_GOAL=""
+fi
+
+if [ -n "$NAMESPACE" ]; then
+    echo "Namespace: $NAMESPACE"
+    MAVEN_OPTS="$MAVEN_OPTS -Dfabric8.namespace=$NAMESPACE"
+    OC_OPTS=" -n $NAMESPACE"
+fi
```

Comment:
Does it make sense to default `$NAMESPACE` to `oc project -q`?

Reply:
Good idea!